tests/Microsoft.Identity.Web.Test/WebApiExtensionsTests.cs

```diff
@@ -0,0 +1,393 @@
+                .AddLogging();
+
+            if (useServiceCollectionExtension)
+                services.AddProtectedWebApi(config, _configSectionName, _jwtBearerScheme, _certificate, subscribeToDiagnostics);
+            else
+                new AuthenticationBuilder(services)
+                    .AddProtectedWebApi(config, _configSectionName, _jwtBearerScheme, _certificate, subscribeToDiagnostics);
+
+            services.RemoveAll<IJwtBearerMiddlewareDiagnostics>();
+            services.AddSingleton<IJwtBearerMiddlewareDiagnostics>((provider) => diagnostics);
+
+            var provider = services.BuildServiceProvider();
+
+            var jwtOptions = provider.GetRequiredService<IOptionsFactory<JwtBearerOptions>>().Create(_jwtBearerScheme);
+
+            if (subscribeToDiagnostics)
+                diagnostics.Received().Subscribe(Arg.Any<JwtBearerEvents>());
+            else
+                diagnostics.DidNotReceive().Subscribe(Arg.Any<JwtBearerEvents>());
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void AddProtectedWebApi_WithConfigActions_SubscribesToDiagnostics(bool useServiceCollectionExtension, bool subscribeToDiagnostics)
+        {
+            var diagnostics = Substitute.For<IJwtBearerMiddlewareDiagnostics>();
+
+            var services = new ServiceCollection()
+                .AddLogging();
+
+            if (useServiceCollectionExtension)
+                services.AddProtectedWebApi(_configureJwtOptions, _configureMsOptions, _certificate, _jwtBearerScheme, subscribeToDiagnostics);
+            else
+                new AuthenticationBuilder(services)
+                    .AddProtectedWebApi(_configureJwtOptions, _configureMsOptions, _certificate, _jwtBearerScheme, subscribeToDiagnostics);
+
+            services.RemoveAll<IJwtBearerMiddlewareDiagnostics>();
+            services.AddSingleton<IJwtBearerMiddlewareDiagnostics>((provider) => diagnostics);
+
+            var provider = services.BuildServiceProvider();
+
+            var jwtOptions = provider.GetRequiredService<IOptionsFactory<JwtBearerOptions>>().Create(_jwtBearerScheme);
+
+            if (subscribeToDiagnostics)
+                diagnostics.Received().Subscribe(Arg.Any<JwtBearerEvents>());
+            else
+                diagnostics.DidNotReceive().Subscribe(Arg.Any<JwtBearerEvents>());
+        }
+
+        private IConfigurationSection GetConfigSection(string configSectionName)
+        {
+            var configAsDictionary = new Dictionary<string, string>()
+            {
+                { configSectionName, null },
+                { $"{configSectionName}:Instance", TestConstants.AadInstance },
+                { $"{configSectionName}:TenantId", TestConstants.TenantIdAsGuid },
+                { $"{configSectionName}:ClientId", TestConstants.TenantIdAsGuid }
```

Comment:
>TenantId [](start = 65, length = 8)

ClientId?

Reply:
I just needed any valid Guid here but I added a TestConstants.ClientId for clarity.